llvm/lib/SYCLLowerIR/TargetHelpers.cpp

```diff
@@ -46,15 +47,74 @@ std::string getAnnotationString(ArchType AT) {
   return std::string();
 }
 
-void populateKernels(Module &M, SmallVectorImpl<KernelPayload> &Kernels,
-                     ArchType AT) {
+SmallPtrSet<Function *, 4> getKernels(Module &M) {
```

Comment:
I'm not sure about this: using `amdgcn.annotations` to identify kernels seems unusual when it's our downstream addition, and they seem to have `amdgcn_kernel` calling conventions. It's how other passes in SYCLLowerIR do it so I blindly copied it. I think I'll try and make it use the calling convention instead, though. Unless there's a good reason for `amdgcn.annotations` I doubt it will make it upstream.

Reply:
See #14713 for a better method.

Reply:
This has gone